### PR TITLE
Fix replacement of stale inodes

### LINKED ISSUE #3481: 2.11.2 crash: InodeTracker.VfsPut
```
#0  0x00007ffff6d05a9f in raise () from /lib64/libc.so.6
#1  0x00007ffff6cd8e05 in abort () from /lib64/libc.so.6
#2  0x00007ffff6cd8cd9 in __assert_fail_base.cold.0 () from /lib64/libc.so.6
#3  0x00007ffff6cfe3f6 in __assert_fail () from /lib64/libc.so.6
#4  0x00007ffff64083e2 in glue::InodeTracker::VfsPutRaii::VfsPut (this=this@entry=0x7fffa6ffccd0, 
    inode=inode@entry=212002, by=185) at /data/build_modules/build/cvmfs-fuse3/cvmfs-fuse3/cvmfs/glue_buffer.h:618
#5  0x00007ffff63fed84 in cvmfs::cvmfs_forget_multi (req=0x7fff98124f90, count=59, forgets=<optimized out>)
    at /data/build_modules/build/cvmfs-fuse3/cvmfs-fuse3/cvmfs/cvmfs.cc:713
#6  0x00007ffff6a9699c in loader::stub_forget_multi (req=0x7fff98124f90, count=59, forgets=0x7fff98000b90)
    at /data/build_modules/build/cvmfs-fuse3/cvmfs-fuse3/cvmfs/loader.cc:333
#7  0x00007ffff6868fea in fuse_session_process_buf_int () from /lib64/libfuse3.so.3
#8  0x00007ffff686439b in fuse_do_work () from /lib64/libfuse3.so.3
#9  0x00007ffff79b31cf in start_thread () from /lib64/libpthread.so.0
#10 0x00007ffff6cf0dd3 in clone () from /lib64/libc.so.6
```

from 

https://github.com/cvmfs/cvmfs/blob/3a5a0f73f14cdbb4d2bd81b7b182d98380538920/cvmfs/glue_buffer.h#L623

From inspecting the core file, the inode number seems sane. 
Could there be a race on forget/removal?
Safe to convert downgrade this from an `assert` to an `AssertOrLog` ?

### PR DESCRIPTION
**Pull request by jblomer**
Fixes #3481 (see issue for details).

Adds testing infrastructure:
  - CVMFS_TEST_INJECT_BARRIER macro
  - "drop metadata caches" talk command

### CODE REVIEW

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest nodestroy

File: test/src/633-changedfiles-corners/main [hunk 1]
```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+# Repreoducer for https://github.com/cvmfs/cvmfs/issues/3481
```
Comment (vvolkl) [inline test/src/633-changedfiles-corners/main]:
```suggestion
# Reproducer for https://github.com/cvmfs/cvmfs/issues/3481
```

**Review by vvolkl — APPROVED** [review]

**Pull request merged**